packages/rrweb/test/record.test.ts

```diff
@@ -586,6 +589,32 @@ describe('record', function (this: ISuite) {
     assertSnapshot(ctx.events);
   });
 
+  it('captures adopted stylesheets of shadow doms in checkout full snapshot', async () => {
+    await ctx.page.evaluate(() => {
+      document.body.innerHTML = `
+        <div id="shadow-host-1">entry</div>
+      `;
+
+      let shadowHost = document.querySelector('div')!;
+      shadowHost!.attachShadow({ mode: 'open' });
+      const sheet = new CSSStyleSheet();
+      sheet.replaceSync!('h1 {color: blue;}');
+      shadowHost.shadowRoot!.adoptedStyleSheets = [sheet];
+
+      const { rrweb, emit } = (window as unknown) as IWindow;
+      rrweb.record({
+        emit,
+      });
+
+      setTimeout(() => {
+        // When a full snapshot is checked out manually, all adoptedStylesheets should also be captured.
+        rrweb.record.takeFullSnapshot(true);
+      }, 10);
+    });
+    await ctx.page.waitForTimeout(200);
+    assertSnapshot(ctx.events);
```

Comment:
This is less flakey because it doesn't depend on `waitForTimeout` (which is also deprecated in newer versions of puppeteer)

```suggestion
    await ctx.page.evaluate(() => {
      return new Promise((resolve) => {
        document.body.innerHTML = `
            <div id="shadow-host-1">entry</div>
          `;

        let shadowHost = document.querySelector('div')!;
        shadowHost!.attachShadow({ mode: 'open' });
        const sheet = new CSSStyleSheet();
        sheet.replaceSync!('h1 {color: blue;}');
        shadowHost.shadowRoot!.adoptedStyleSheets = [sheet];

        const { rrweb, emit } = (window as unknown) as IWindow;
        rrweb.record({
          emit,
        });

        setTimeout(() => {
          // When a full snapshot is checked out manually, all adoptedStylesheets should also be captured.
          rrweb.record.takeFullSnapshot(true);
          resolve(undefined);
        }, 10);
      });
    });
    await waitForRAF(page);
    assertSnapshot(ctx.events);
```

Reply:
Thanks for the suggestion! I will keep this in mind for later test cases.